src/java/org/apache/cassandra/locator/PendingRangeMaps.java

```diff
@@ -23,196 +23,176 @@
+                replicasToAdd.addAll(replicas);
             }
         }
     }
 
-    public Collection<InetAddressAndPort> pendingEndpointsFor(Token token)
+    public ReplicaSet pendingEndpointsFor(Token token)
     {
-        Set<InetAddressAndPort> endpoints = new HashSet<>();
+        ReplicaSet replicas = new ReplicaSet();
 
-        Range searchRange = new Range(token, token);
+        Range<Token> searchRange = new Range<>(token, token);
 
         // search for non-wrap-around maps
-        NavigableMap<Range<Token>, List<InetAddressAndPort>> ascendingTailMap = ascendingMap.tailMap(searchRange, true);
-        NavigableMap<Range<Token>, List<InetAddressAndPort>> descendingTailMap = descendingMap.tailMap(searchRange, false);
+        NavigableMap<Range<Token>, List<Replica>> ascendingTailMap = ascendingMap.tailMap(searchRange, true);
+        NavigableMap<Range<Token>, List<Replica>> descendingTailMap = descendingMap.tailMap(searchRange, false);
 
         // add intersections of two maps
         if (ascendingTailMap.size() < descendingTailMap.size())
         {
-            addIntersections(endpoints, ascendingTailMap, descendingTailMap);
+            addIntersections(replicas, ascendingTailMap, descendingTailMap);
         }
         else
         {
-            addIntersections(endpoints, descendingTailMap, ascendingTailMap);
+            addIntersections(replicas, descendingTailMap, ascendingTailMap);
         }
 
         // search for wrap-around sets
         ascendingTailMap = ascendingMapForWrapAround.tailMap(searchRange, true);
         descendingTailMap = descendingMapForWrapAround.tailMap(searchRange, false);
 
         // add them since they are all necessary.
-        for (Map.Entry<Range<Token>, List<InetAddressAndPort>> entry : ascendingTailMap.entrySet())
+        for (Map.Entry<Range<Token>, List<Replica>> entry : ascendingTailMap.entrySet())
         {
-            endpoints.addAll(entry.getValue());
+            replicas.addAll(entry.getValue());
         }
-        for (Map.Entry<Range<Token>, List<InetAddressAndPort>> entry : descendingTailMap.entrySet())
+        for (Map.Entry<Range<Token>, List<Replica>> entry : descendingTailMap.entrySet())
         {
-            endpoints.addAll(entry.getValue());
+            replicas.addAll(entry.getValue());
         }
 
-        return endpoints;
+        return replicas;
     }
 
     public String printPendingRanges()
     {
         StringBuilder sb = new StringBuilder();
 
-        for (Map.Entry<Range<Token>, List<InetAddressAndPort>> entry : this)
+        for (Map.Entry<Range<Token>, List<Replica>> entry : this)
         {
             Range<Token> range = entry.getKey();
 
-            for (InetAddressAndPort address : entry.getValue())
+            for (Replica replica : entry.getValue())
             {
-                sb.append(address).append(':').append(range);
+                sb.append(replica).append(':').append(range);
```

Comment:
Some, I'm sure. Not sure what we can really do about it though, since we're adding the port and transient data it kind of needs to be included here. At least it's in a major release though?